static/js/util/profile_edit.js

```diff
@@ -147,146 +137,15 @@ export function boundDateField(keySet: string[], label: string, omitDay: boolean
-    // will have text at a time since the user can't edit more than one field at once
-    // so we need to look in the state to see
-    let newEdit = Object.assign({}, edit, {
-      year: year !== undefined ? year : edit.year,
-      month: month !== undefined ? month : edit.month,
-      day: day !== undefined ? day : edit.day
-    });
-
-    let validatedDay = Maybe.of(1);
-    if (!omitDay) {
-      validatedDay = validateDay(newEdit.day);
-      newEdit.day = mstr(validatedDay);
-    }
-
-    let validatedMonth = validateMonth(newEdit.month);
-    newEdit.month = mstr(validatedMonth);
-
-    let validatedYear = validateYear(newEdit.year);
-    newEdit.year = mstr(validatedYear);
-
-    // keep text up to date
-    _.set(clone, editKeySet, newEdit);
-
-    let padYear = s => _.padStart(s, 4, '0');
-
-    let dateList = [validatedYear, validatedMonth, validatedDay];
-
-    let stringifyDates = R.compose(
-      R.join("-"), R.map(mstr), R.adjust(x => x.map(padYear), 0)
-    );
-
-    let dateString = S.maybe("", stringifyDates, allJust(dateList));
-
-    let rawDate = Maybe.of(moment(dateString, ISO_8601_FORMAT));
-
-    let validatedDate = rawDate.filter(date => date.isValid()).
-      filter(date => date.isAfter(moment("1800", "YYYY")));
-
-    if ( validatedDate.isNothing ) {
-      _.set(clone, keySet, null);
-    } else {
-      _.set(clone, keySet, validatedDate.value.format(ISO_8601_FORMAT));
-    }
-    updateProfile(clone, validator);
-  };
-
-  let edit = getEditObject();
-
-  let dayField, daySlash;
-  if (!omitDay) {
-    daySlash = <span className="slash"> / </span>;
-    dayField = <TextField
-      hintText="DD"
-      floatingLabelText=" "
-      floatingLabelFixed={true}
-      style={{
-        maxWidth: "2em"
-      }}
-      fullWidth={true}
-      value={edit.day !== undefined ? edit.day : ""}
-      onChange={e => setNewDate(e.target.value, undefined, undefined)}
-    />;
-  }
-
-  return <div className={validationErrorSelector(errors, keySet)}>
-    <TextField
-      floatingLabelText={label}
-      floatingLabelFixed={true}
-      floatingLabelStyle={{whiteSpace: "nowrap"}}
-      hintText="MM"
-      style={{
-        maxWidth: "2em"
-      }}
-      errorStyle={{
-        position: "absolute",
-        top: "100%",
-        whiteSpace: "nowrap"
-      }}
-      fullWidth={true}
-      value={edit.month !== undefined ? edit.month : ""}
-      onChange={e => setNewDate(undefined, e.target.value, undefined)}
-      errorText={_.get(errors, keySet)}
-    />
-    <span className="slash"> / </span>
-    {dayField}
-    {daySlash}
-    <TextField
-      hintText="YYYY"
-      floatingLabelFixed={true}
-      floatingLabelText=" "
-      style={{
-        maxWidth: "4em"
-      }}
-      fullWidth={true}
-      value={edit.year !== undefined ? edit.year : ""}
-      onChange={e => setNewDate(undefined, undefined, e.target.value)}
-    />
-  </div>;
+  return <DateField
+    data={profile}
```

Comment:
not wild about a prop as ambiguous as `data`, but truth be told, i don't think it was any more sensible before this refactor. we can look into this post-deadline